server/api/user/update/service.js

```diff
@@ -0,0 +1,126 @@
+import { ObjectId } from 'mongodb';
```

Comment:
Some notes here affecting the whole service:

1. You may export multiple functions per service if necessary (just FYI).
2. As a general rule of thumb, you do not need to test authorization logic with services. The service is oblivious to whether the user is superadmin or not, whether they have the right permissions or not. It's the controller's job to ensure authorization is valid. That said, the service will only make sure the data provided make sense (e.g. username does not already exist) and process the request.
3. Important: this is a very long method. Worth exploring imo if it could be simplified using the decorator pattern to set user props gradually (or throw error if need be) and then commit changes to the db. Makes sense?